Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
Sure, I'm not questioning that, just raising an eyebrow at the naming : your method should be called `setDoctrineRegistry` instead of `setEntityManager`, shouldn't it?